pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -732,8 +779,10 @@ impl RemoteTimelineClient {
             // upload finishes or times out soon enough.
             if task_mgr::is_shutdown_requested() {
                 info!("upload task cancelled by shutdown request");
+                if let Err(e) = self.stop() {
+                    error!("got an error when trying to stop remote client: {e}")
+                }
                 self.calls_unfinished_metric_end(&task.op);
-                self.stop();
                 return;
```

Comment:
The `error!()` doesn't bail out. We effectively ignore the failure to stop.

Looking at the failure modes of self.stop(), I think we should panic here.

Also, it would help if self.stop() returned distinguished error types, and we could write a match block with unreachable!() clauses, explaining why neither failure is unreachable.

Alternative to panicking: ignore the shutdown request, and log that.

But, we shouldn't just return in the case where stop() fails.

Reply:
Fixed this in https://github.com/neondatabase/neon/pull/3919/commits/11c18b05aa67ecbe3ed3db2ae92e60da86d3bd31